airsonic-main/src/main/java/org/airsonic/player/upload/MonitoredMultipartFile.java

```diff
@@ -67,10 +68,9 @@ public InputStream getInputStream() throws IOException {
 
     @Override
     public void transferTo(File dest) throws IOException, IllegalStateException {
-        try (
-            FileOutputStream fos = new FileOutputStream(dest);
-            MonitoredOutputStream monitoredOutputStream = new MonitoredOutputStream(fos, listener)) {
-            monitoredOutputStream.write(file.getBytes());
-        }
+        InputStream inputStream = file.getInputStream();
+        FileOutputStream outputStream = new FileOutputStream(dest);
+        MonitoredOutputStream monitoredOutputStream = new MonitoredOutputStream(outputStream, listener);
+        FileCopyUtils.copy(inputStream, monitoredOutputStream);
```

Comment:
@zc-devs 
Thank you for your contribution!
I noticed you're removing the try-with-resource statement for the OutputStream. This ensures that the stream is automatically closed, so there's no need for an explicit close() call. Please make sure not to remove the try-with-resource usage, as it's a more efficient and safer way to handle resource management.

```suggestion
        InputStream inputStream = file.getInputStream();
        try (
            FileOutputStream outputStream = new FileOutputStream(dest);
            MonitoredOutputStream monitoredOutputStream = new MonitoredOutputStream(outputStream, listener)) {
                FileCopyUtils.copy(inputStream, monitoredOutputStream);
         }
```

Reply:
org.springframework.util.FileCopyUtils#copy(java.io.InputStream, java.io.OutputStream) already closes both streams.

> Copy the contents of the given InputStream to the given OutputStream. Closes both streams when done.